### PR TITLE
I18n(es_es)

### PR DESCRIPTION
**Pull request by joebordes**
A draft version of the ES Spain translation.

All the (close to 1700) labels are translated and a first revision of the translated strings and from inside the application is done.

It needs more validation and there are still some strings that have not been internationalized in the application so we will have to add those here as they get i18ned.

I share this here so others can help and construct upon this work.

bienvenidometabase

### CODE REVIEW

**Comment by joebordes** [comment]
Any chance we can get #7062, #7063, #7064 and this PR merged so I can have another work session and push this forward. See if we can get it in for the next release.

Thanks

**Comment by joebordes** [comment]
bump

**Comment by joebordes** [comment]
Perfect. Thanks!
I'm going to give the es translation another go as soon as I can.

**Comment by tentocs** [comment]
@joebordes 

how could you do it ?, I try to execute this "/ bin / i18n / update-translations" and it generates me error


SyntaxError: frontend/src/metabase/admin/databases/components/CreatedDatabaseModal.jsx: You can not use JSXElement

**Comment by joebordes** [comment]
have you launched the other "yarn" commands necessary for development?

mostly 

```
yarn
```

to download all the dependencies.

I will try to get back to this soon and try again to confirm it works

**Comment by tentocs** [comment]
What are the commands that you execute every time you make a change to Spanish?

I already installed all the dependencies of yarn but it still generates the same error and it does not update anything in the view

**Comment by joebordes** [comment]
I just dedicated some more time to reviewing and updating the ES translation. I had no problem launching the update command and it worked correctly.

When I set up the development environment I remember having to launch yarn, yarn build, build and similar various times until I got it right. All the commands and steps are [on the documentation page](https://metabase.com/docs/latest/developers-guide.html).

What I do now is execute ```bin/i18n/update-translations``` to get the new strings. I translate and then I build with these two commands:

```
yarn run build-hot
lein ring server
```

each on different terminals.
I am developing on linux

HTH

**Comment by mauricio-sonegatti** [comment]
@joebordes I'm having problems with 5 digit language codes ex:  pt-br, pt-pt I can create the po but it will not build because of the "-" on the class name, I know I saw somewhere you discussing about that, but I guess the code doesn't support it yet right?

**Comment by joebordes** [comment]
as far as I can see they did not take make comments into consideration, you will have to do pt only
I suppose they will revisit this issue again in the future
At least we have a good start.

File: locales/es.po [hunk 1]
```diff
@@ -0,0 +1,8574 @@
...
+
+#: frontend/src/metabase/admin/settings/components/SettingsEmailForm.jsx:75
+#: frontend/src/metabase/admin/settings/components/SettingsLdapForm.jsx:72
+#: frontend/src/metabase/admin/settings/components/SettingsSlackForm.jsx:67
+#: frontend/src/metabase/admin/settings/selectors.js:150
+msgid "That's not a valid email address"
+msgstr "Esa no es una dirección de correo electrónico válida"
+
+#: frontend/src/metabase/admin/settings/components/SettingsEmailForm.jsx:79
+#: frontend/src/metabase/admin/settin
```
Comment (mazameli) [inline locales/es.po]: Looks like this should be an `o` instead of an `@`

**Comment by mazameli** [comment]
Sorry for the silence, @joebordes. I'm going to look at this right now and see if we can get this merged asap.

**Comment by joebordes** [comment]
Thanks @mazameli 

I received an email notification saying "Looks like this should be an o instead of an @" but I can't find this comment anywhere in the PR. I suppose it is an internal review comment but if it isn't please explain and I will try to fix. If I can help in any way let me know.

**Comment by mazameli** [comment]
Thanks, Joe. Yeah, I had put that comment in re: the use of `Bienvenid@`, not realizing that that's apparently a modern way of saying `bienvenido` + `bienvenida`. It struck us as slightly odd looking, but then again, none of us are native Spanish speakers.

Beyond that, I'm just going through the app using this branch and making sure things look right. We already knew there were some backend user-facing strings that haven't been properly tagged yet, so we're working on that as well, and I'll let you know once those are in place.

**Comment by joebordes** [comment]
sounds great :-)
I'm listening

**Comment by tentocs** [comment]
@joebordes  
I have done what you told me but I have not been able to solve it, reinstall everything and still follow the same error, which is the same one that was presented to you, look:

```
./bin/i18n/update-translations
SyntaxError: frontend/src/metabase/admin/settings/components/SettingsSingleSignOnForm.jsx: You can not use JSXElement '${<a className="link" href="https://developers.google.com/identity/sign-in/web/devconsole-project" target="_blank">
                here.
              </a>}' in localized strings
  117 |           </p>
  118 |           <p className="text-grey-4">
> 119 |             {jt`To allow users to sign in with Google you'll need to give Metabase a Google Developers console application client ID. It only takes a few steps and instructions on how to create a key can be found ${(
      |              ^
  120 |               <a
  121 |                 className="link"
  122 |                 href="https://developers.google.com/identity/sign-in/web/devconsole-project"
```

**Comment by joebordes** [comment]
@tentocs I'm sorry but I don't know what else to say. I just tried this again, as you say, download a new version, yarn, and update translations with no problem (I am on a different machine for this test).

**Comment by mazameli** [comment]
Hey, @joebordes! So I went through and checked all of the strings that didn't appear to be translated into Spanish to see if that was because we hadn't wrapped them, or if they just had not yet been translated.

I found 3 cases where we'd overlooked the i18n wrapping, here: https://github.com/metabase/metabase/pull/7372/commits/a58d5e4b3b6cee881f3e4c2d3baf96260fa79c60

[The rest of the cases I found are screenshotted here](https://github.com/metabase/metabase/issues/6674#issuecomment-382155179), and they all appear to me to be correctly present in the .pot file, and just need translation. Does that appear to be correct to you?

**Comment by mazameli** [comment]
Awesome, thanks! I forgot to add that [in this PR](https://github.com/metabase/metabase/pull/7372) we're adding i18n wrapping for the query builder datetime grouping options (day, week, hour, etc.), binning options, plus a bunch of backend strings that we missed.

**Comment by joebordes** [comment]
You mean I am in for another session of a lot of fun :-D

**Comment by mazameli** [comment]
Haha, all I can say is thank you!

**Comment by tentocs** [comment]
@joebordes 
thanks for the recommendation, I downloaded the version where you could translate, run `./bin/i18n/update-translations` and I do not generate error, continue with the other steps but did not translate.

**Comment by joebordes** [comment]
@tentocs I'm glad you got it working :-)

**Comment by mazameli** [comment]
Hey, Joe. We just merged https://github.com/metabase/metabase/pull/7372 into master to externalize the backend strings and few front-end ones we'd previously missed.

**Comment by joebordes** [comment]
Ok, I'm going to get those. I just finished updating the translation a while ago and I was testing it. It is looking REALLY good.

The only important thing I could see is that I cannot access setup and authentication in settings. I will update with the new strings and get back to you soon.

Thanks!

**Comment by mauricio-sonegatti** [comment]
Hey @joebordes , are you having any problems with displaying the translated language in the new version?
The code is working ok for yesterdays commits, but the system is not showing the translations.

**Comment by joebordes** [comment]
Hi @mauricio-sonegatti 

Truth is I haven't tried it in a while, I am losing interest as this is taking WAY too long (it is almost a year now and I have tried really hard to make it happen). I mean i18n support isn't even on the roadmap for 0.30...

Just to answer your question I updated my local repository, launched yarn and then tried to update the translation files, but got an error: #7966

I will try it again when they fix it, update the translation and give it a try. I'll let you know how it goes.

**Comment by mazameli** [comment]
Hi, @joebordes. I sent you an email a couple days ago to this effect, but in case you missed it:

We wanted to apologize for the lack of communication on our end with regard to the timing and planning of merging and shipping this PR, and we're very grateful for all the work you've put into it.

We are putting the finishing touches on the 0.30 release, but we would love to get this PR in it if you're still willing and able to update it, along with the new/updated strings in 0.30. We did manage to fix #7966 (which was a dupe of https://github.com/metabase/metabase/issues/7957).

Also, going forward we're considering using [POEditor](https://poeditor.com/) to make it much easier for the community to contribute to translations (especially less technical folks), and we'd be interested in your thoughts about that.

Thanks again so much, and we look forward to merging this and getting Metabase in Spanish!

**Comment by joebordes** [comment]
Hi @mazameli , sorry I missed that email.

An apology is not needed, I understand and I see that not only are you keeping busy but you are keeping the translation effort going which is the important part of this.

I have updated the es translation to the latest strings as of today and I have created two tickets with the issues I have found. There is at least one blocker depending on your level of demand.

As to poeditor, transiflex or any other of the many tools like that out there. I have used them in the past as each project seems to use a different one and I like spending time translating open source projects. They are all mostly the same and they help to some extent. Personally, I am a coder and I am really comfortable editing the files directly but I have no problem with you using poeditor or any other.

Let me know if there is anything I can help with.

**Comment by mazameli** [comment]
Thanks so much, @joebordes!

**Comment by tlrobinson** [comment]
Here's something we should address before launching localization: #8209

**Comment by tlrobinson** [comment]
@joebordes Thanks for all your hard work on this!

**Comment by joebordes** [comment]
My pleasure :-)

I'll give it a try this week. Thanks!!

**Pull request merged**